owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1438,29 +1443,41 @@ class FileDisplayActivity : FileActivity(),
      * @param file [OCFile] file which operation are wanted to be cancel
      */
     fun cancelTransference(file: OCFile) {
-        transfersViewModel.cancelTransfersForFile(file)
-        fileWaitingToPreview?.let {
-            if (it.remotePath == file.remotePath) {
-                fileWaitingToPreview = null
+        if (file.isFolder) {
+            val result = runBlocking(CoroutinesDispatcherProvider().io) {
+                getFolderContentUseCase.execute(GetFolderContentUseCase.Params(file.id!!))
+            }
+            val files = result.getDataOrNull()
+            files?.let {
+                it.forEach { file ->
+                    cancelTransference(file)
+                }
+            }
```

Comment:
I don't think so, since we need to call recursively this function (which performs also some UI tasks). We could move to the ViewModel the part where we retrieve the folder content, but I didn't because it doesn't fit very well in the TransfersViewModel (we are retrieving files, not transfers).

Reply:
Ok, I performed some changes to be able to move it to the ViewModel 😀